src/main/java/seedu/address/seplendidui/MainWindow.java

```diff
@@ -0,0 +1,145 @@
+
+import javafx.beans.binding.Bindings;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.scene.layout.StackPane;
+import javafx.stage.Stage;
+import seedu.address.commons.core.GuiSettings;
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.logic.Logic;
+import seedu.address.logic.SeplendidLogic;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.SeplendidDataType;
+import seedu.address.model.localcourse.LocalCourse;
+
+/**
+ * The Main Window. Provides the basic application layout containing
+ * a menu bar and space where other JavaFX elements can be placed.
+ */
+public class MainWindow extends UiPart<Stage> {
+
+    private static final String FXML = "MainWindow.fxml";
+
+    private final Logger logger = LogsCenter.getLogger(getClass());
+
+    private Stage primaryStage;
+    private SeplendidLogic seplendidLogic;
+
+    // Independent Ui parts residing in this Ui container
+    private ItemListPanel<SeplendidDataType> itemListPanel;
+    private ItemDetailPanel<SeplendidDataType> itemDetailPanel;
+
+    private ResultBox resultBox;
+
+    @FXML
+    private StackPane commandBoxPlaceholder;
+
+    @FXML
+    private StackPane resultBoxPlaceholder;
+
+    @FXML
+    private StackPane itemListPanelPlaceholder;
+
+    @FXML
+    private StackPane itemDetailPanelPlaceholder;
+
+
+    /**
+     * Creates a {@code MainWindow} with the given {@code Stage} and {@code Logic}.
+     */
+    public MainWindow(Stage primaryStage, SeplendidLogic seplendidLogic) {
+        super(FXML, primaryStage);
+
+        // Set dependencies
+        this.primaryStage = primaryStage;
+        this.seplendidLogic = seplendidLogic;
+
+        // Configure the UI
+        setWindowDefaultSize(seplendidLogic.getGuiSettings());
+    }
+
+    public Stage getPrimaryStage() {
+        return primaryStage;
+    }
+
+    /**
+     * Fills up all the placeholders of this window.
+     */
+    void fillInnerParts() {
+        itemDetailPanel = new ItemDetailPanel<>();
+        itemDetailPanelPlaceholder.getChildren().add(itemDetailPanel.getRoot());
+
+        ObservableList<LocalCourse> localList = seplendidLogic.getFilteredLocalCourseCatalogue();
+        ObservableList<SeplendidDataType> itemList = FXCollections.observableArrayList(localList);
+        Bindings.bindContent(itemList, localList);
+
+        itemListPanel = new ItemListPanel<>(itemList, itemDetailPanel);
+        itemListPanelPlaceholder.getChildren().add(itemListPanel.getRoot());
+
+        resultBox = new ResultBox();
+        resultBoxPlaceholder.getChildren().add(resultBox.getRoot());
+
+        CommandBox commandBox = new CommandBox(this::executeCommand);
+        commandBoxPlaceholder.getChildren().add(commandBox.getRoot());
+    }
```

Comment:
Note: TODO: Need to add support, and switching between different `ObservableList`s for different commands.